src/newrelic_telemetry_sdk/client.py

```diff
@@ -227,3 +227,48 @@ class MetricClient(Client):
     HOST = "metric-api.newrelic.com"
     URL = "/metric/v1"
     PAYLOAD_TYPE = "metrics"
+
+
+class EventClient(Client):
+    """HTTP Client for interacting with the New Relic Event API
+
+    This class is used to send events to the New Relic Event API over HTTP.
+
+    :param insert_key: Insights insert key
+    :type insert_key: str
+    :param host: (optional) Override the host for the event API
+        endpoint.
+    :type host: str
+    :param compression_threshold: (optional) Compress if number of bytes in
+        payload is above this threshold. (Default: 64K)
+    :type compression_threshold: int
+
+    Usage::
+
+        >>> import os
+        >>> insert_key = os.environ.get("NEW_RELIC_INSERT_KEY", "")
+        >>> event_client = EventClient(insert_key)
+        >>> response = event_client.send({})
+    """
+
+    HOST = "insights-collector.newrelic.com"
+    URL = "/v1/accounts/events"
+
+    def send_batch(self, items):
```

Comment:
This method has a different signature than the parent class.
We should probably allow for including an optional common block when sending a list of events. Right now the logic for including the common attributes with the events lives in the EventBatch with the common attributes being inserted on flush. If in the future ingest updates to take a single common block for a batch of events we should only have to update the client. 

It is also almost all duplicate logic. 
Maybe we make a get_payload(items, common) which will build the json object for the specific client type? We can then override that method. Or we can abstract the send logic into a private method something like `def _send(self, payload)`  which handles the encoding, compressing, and posting of data.

Reply:
Since the common tags provided in the batch constructor are always _user attributes_, never intrinsics, I don't think they are as problematic.  
Also, I find the fact that the `EventBatch` constructor signature doesn't match the other batch constructors to be more jarring than the difference in the send_batch signature.

Reply:
The `tags` parameter means something different between EventBatch and other BatchTypes. For other batch types `tags` are included as nested attributes, and even if they are eventually flattened they are still sent. If we include optional common attributes for the EventBatch we would have to flatten them into the payload as in the sdk. Doing this manipulation in the client to account for a feature the backend does not support yet could lead to major forward compatibility issues. My vote is to release the first iteration without support for a common block for custom events.